### DISCUSSION TITLE
Problems installing ruff and sqlfluff

### DISCUSSION THREAD

**Discussion by grifferz** [mason-org/mason.nvim#1823] [Q&A] [answered]
I keep my neovim config synced between hosts. I started doing some Python work on one host so did MasonInstall and got ruff. Now on another host at every start of neovim I get an error about failing to install ruff.

Here is the error

```
[INFO Thu 31 Oct 2024 21:48:28 UTC] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184:                   Executing installer for Package(name=sqlfluff) {}                                                              
[INFO Thu 31 Oct 2024 21:48:28 UTC] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184:                   Executing installer for Package(name=ruff) {}                                                                  
[ERROR Thu 31 Oct 2024 21:48:28 UTC] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:249:                   Installation failed for Package(name=ruff) error=spawn: python3 failed with exit code 1 and signal 0.·         
[ERROR Thu 31 Oct 2024 21:48:28 UTC] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:249:                   Installation failed for Package(name=sqlfluff) error=spawn: python3 failed with exit code 1 and signal 0.
```

I am able to `pip install sqlfluff` (and same for `ruff`) on that system both for my user and system wide with `sudo`. Is there a way to get more information on what is failing for Mason exactly?

**Comment by Gentil-N** [comment, marked as answer]
Hi @grifferz,
While using Lazyvim, I faced the same issue for ruff and solved it this way:

1. Type `:MasonInstall ruff`
2. Pull down the menu associated to ruff and look at the error message. For me, the package "virtual environment" was missing.
3. **Only for Ubuntu environments** Enter in a terminal `sudo apt install python3-venv`
4. Restart nvim and enter `:MasonInstall ruff` again.

Not sure if all the steps are necessary as I am a newbie in neovim but it worked. Good luck!

**Reply by grifferz** [reply]
Thank you _so_ much! For me also it wanted `sudo apt install python3-venv`.

**Reply by thomasrpo03** [reply]
Thanks a lot! i had the same issue

**Reply by geovannewashington** [reply]
You are my hero

**Reply by sigurdpl** [reply]
same probably for me on ubuntu vm. Missing python virtual environment.

**Reply by eric-burel** [reply]
Hi, still digging this issue when installing Python with "uv", python3-venv was installed. Error is `['/home/eric-burel/.local/share/nvim/mason/packages/ruff/venv/bin/python3', '-m', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1` so I guess this is a similar issue though, pip is not found somehow. It seems I must fix this error first : `python -m ensurepip # error: externally-managed-environment`. Edit: tracked here https://github.com/astral-sh/uv/issues/18204
